### ISSUE TITLE
Add support for report floorTemperature

### ISSUE THREAD

**Issue by roscoegray** [MindrustUK/Heatmiser-for-home-assistant#72]
Cheeky request to add support to report the floor probe temperature, if and when someone has the time and desire.

I used to use this in OpenHab, before switching over to HA, to trigger a temporary rise to the set level if the floor temp dropped below a threshold. It was great as a few rooms wouldn't get cold enough to need warming, but the floor would get cold as the floor is suspended.

I'm just a hobbyist coder so don't think I can get my head around how it was done, looks to me like the engineer's data was used...
https://github.com/openhab/openhab-addons/tree/main/bundles/org.openhab.binding.neohub

**Comment by stikonas** [comment]
it should be quite easy to add. In fact I think you should be able to do it even with hobbyist coding skills.

`neohubapi` library already reads `current_floor_temperature` attribute. So in this integration it should be accessible via `self._neostat.current_floor_temperature` in `climate.py`.

**Comment by stikonas** [comment]
> can this be done in the climate entity, as it doesn't support an additional temperature property? I wouldn't want to swap current_temperature for current_floor_temperature, I would want to report them both. Would another entity be needed, like sensor?
> I will have a look through the documentation and see if I can understand it. Thanks for your reply.

Hmm, good point. Yes it should go into a separate sensor. I guess that means getting this integration to recognize Devices (https://developers.home-assistant.io/docs/device_registry_index) is getting more important.... Then climate entiry and floor temporature sensors can be assigned to the same device.

**Comment by roscoegray** [comment]
I had a go [here](https://github.com/roscoegray/Heatmiser-for-home-assistant), and now I have all the floor readings from each stat as a sensor, not that everyone has a sensor floor prob so could do with some logic to check for that. Makes sense for the integration to register the stats as devices so they are assigned together, it took me some time to figure out the sensor bit, I will continue to read up and give it a go.

**Comment by stikonas** [comment]
> I had a go [here](https://github.com/roscoegray/Heatmiser-for-home-assistant), and now I have all the floor readings from each stat as a sensor, not that everyone has a sensor floor prob so could do with some logic to check for that. Makes sense for the integration to register the stats as devices so they are assigned together, it took me some time to figure out the sensor bit, I will continue to read up and give it a go.

So I don't have any floor sensor and current floor temperature here reads

`current_floor_temperature: 127.5`. Maybe this value can be used to indicate lack of sensor. Can anybody else without floor sensor confirm?

**Comment by roscoegray** [comment]
Yes I get the same for my neo stats that don't have floor probes, so yes I would think this could be used to indicate no sensor connected.

**Comment by PhillyGilly** [comment]
I get current_floor_temperature as 127 in my debug log for devices without a temperature sensor
06.	{'AWAY': False, 'COOLING': False, 'COOLING_ENABLED': False, 'COOLING_TEMPERATURE_IN_WHOLE_DEGREES': 0, 'COOL_INP': False, 'COUNT_DOWN_TIME': '0:00', 'CRADLE_PAIRED_TO_REMOTE_SENSOR': False, 'CRADLE_PAIRED_TO_STAT': False, '**CURRENT_FLOOR_TEMPERATURE': 127,** 'CURRENT_SET_TEMPERATURE': '20.0', 'CURRENT_TEMPERATURE': '19.2', 'DEMAND': False, 'DEVICE_TYPE': 1, 'ENABLE_BOILER': False, 'ENABLE_COOLING': False, 'ENABLE_PUMP': False, 'ENABLE_VALVE': False, 'ENABLE_ZONE': False, 'FAILSAFE_STATE': False, 'FAIL_SAFE_ENABLED': False, 'FLOOR_LIMIT': False, 'FULL/PARTIAL_LOCK_AVAILABLE': False, 'HEAT/COOL_MODE': False, 'HEATING': False, 'HOLD_TEMPERATURE': 17, 'HOLD_TIME': '0:00', 'HOLIDAY': False, 'HOLIDAY_DAYS': 0, 'HUMIDITY': 0, 'LOCK': False, 'LOCK_PIN_NUMBER': '0000', 'LOW_BATTERY': False, 'MAX_TEMPERATURE': '24.0', 'MIN_TEMPERATURE': '20.0', 'MODULATION_LEVEL': 0, 'NEXT_ON_TIME': '2 days 22:00', 'OFFLINE': False, 'OUPUT_DELAY': False, 'OUTPUT_DELAY': 0, 'PREHEAT': False, 'PREHEAT_TIME': '255:255', 'PROGRAM_MODE': '5DAY/2DAY', 'PUMP_DELAY': False, 'RADIATORS_OR_UNDERFLOOR': False, 'SENSOR_SELECTION': 'BUILT_IN_AIR_SENSOR', 'SET_COUNTDOWN_TIME': 0, 'STANDBY': False, 'STAT_MODE': {'4_HEAT_LEVELS': True, 'MANUAL_OFF': True, 'THERMOSTAT': True}, 'TEMPERATURE_FORMAT': False, 'TEMP_HOLD': False, 'TIMECLOCK_MODE': False, 'TIMER': False, 'TIME_CLOCK_OVERIDE_BIT': False, 'ULTRA_VERSION': 0, 'VERSION_NUMBER': 101, 'WRITE_COUNT': 0, 'ZONE_1PAIRED_TO_MULTILINK': True, 'ZONE_1_OR_2': False, 'ZONE_2_PAIRED_TO_MULTILINK': False, 'device': 'Bedroom 2'}, 
07.	{'AWAY': False, 'COOLING': False, 'COOLING_ENABLED': False, 'COOLING_TEMPERATURE_IN_WHOLE_DEGREES': 0, 'COOL_INP': False, 'COUNT_DOWN_TIME': '0:00', 'CRADLE_PAIRED_TO_REMOTE_SENSOR': False, 'CRADLE_PAIRED_TO_STAT': False, **'CURRENT_FLOOR_TEMPERATURE': 22**, 'CURRENT_SET_TEMPERATURE': '17.0', 'CURRENT_TEMPERATURE': '21.1', 'DEMAND': False, 'DEVICE_TYPE': 1, 'ENABLE_BOILER': False, 'ENABLE_COOLING': False, 'ENABLE_PUMP': False, 'ENABLE_VALVE': False, 'ENABLE_ZONE': False, 'FAILSAFE_STATE': False, 'FAIL_SAFE_ENABLED': False, 'FLOOR_LIMIT': False, 'FULL/PARTIAL_LOCK_AVAILABLE': False, 'HEAT/COOL_MODE': False, 'HEATING': False, 'HOLD_TEMPERATURE': 22, 'HOLD_TIME': '0:00', 'HOLIDAY': False, 'HOLIDAY_DAYS': 0, 'HUMIDITY': 0, 'LOCK': False, 'LOCK_PIN_NUMBER': '0000', 'LOW_BATTERY': False, 'MAX_TEMPERATURE': '23.0', 'MIN_TEMPERATURE': '21.0', 'MODULATION_LEVEL': 0, 'NEXT_ON_TIME': '2 days 21:00', 'OFFLINE': False, 'OUPUT_DELAY': False, 'OUTPUT_DELAY': 0, 'PREHEAT': False, 'PREHEAT_TIME': '255:255', 'PROGRAM_MODE': '5DAY/2DAY', 'PUMP_DELAY': False, 'RADIATORS_OR_UNDERFLOOR': False, 'SENSOR_SELECTION': 'REMOTE_AIR_SENSOR_AND_FLOOR', 'SET_COUNTDOWN_TIME': 0, 'STANDBY': False, 'STAT_MODE': {'4_HEAT_LEVELS': True, 'MANUAL_OFF': True, 'THERMOSTAT': True}, 'TEMPERATURE_FORMAT': False, 'TEMP_HOLD': False, 'TIMECLOCK_MODE': False, 'TIMER': False, 'TIME_CLOCK_OVERIDE_BIT': False, 'ULTRA_VERSION': 0, 'VERSION_NUMBER': 101, 'WRITE_COUNT': 10, 'ZONE_1PAIRED_TO_MULTILINK': True, 'ZONE_1_OR_2': False, 'ZONE_2_PAIRED_TO_MULTILINK': False, 'device': 'Bathroom 2'},

**Comment by roscoegray** [comment]
Interesting. @PhillyGilly, do you have a V1 or V2 neohub? The V2's had an update last year enabling 0.5 setpoint control. I wonder if that is why mine and stikonas report 127.5 and you are getting 127?

**Comment by PhillyGilly** [comment]
> Interesting. @PhillyGilly, do you have a V1 or V2 neohub? The V2's had an update last year enabling 0.5 setpoint control. I wonder if that is why mine and stikonas report 127.5 and you are getting 127?

Yes. My neohub is a V1.

**Comment by roscoegray** [comment]
That kind of makes sense then. If a condition is implemented using this value it will need to keep this in mind for compatibility of the V1 neohub.

**Comment by roscoegray** [comment]
Submitted PR which addresses this #83

**Comment by hollytreecook** [comment]
How do I get to see the NeoStat temperatures in rooms. I have 16 thermostats on NeoHub V2.
I can see 16 devices and 16 entities, but they all appear to be swithches.
Thanks Steve
![Screenshot 2023-04-25 121400](https://user-images.githubusercontent.com/131459314/234263589-44e55548-1967-4353-8c84-a40632b17558.png)

**Comment by MindrustUK** [comment]
@hollytreecook What thermostats do you have exactly? I've seen some issues upstream with device types which is on my todo list. See: https://github.com/MindrustUK/Heatmiser-for-home-assistant/issues/121 for further details.

**Comment by MindrustUK** [comment]
@hollytreecook Also please open a new issue for questions, thread necromancy is discouraged :)
